xray/usage/reportusage_test.go

```diff
@@ -0,0 +1,132 @@
+package usage
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/jfrog/jfrog-client-go/utils"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestIsXrayVersionCompatible(t *testing.T) {
+	tests := []struct {
+		xrayVersion    string
+		expectedResult bool
+	}{
+		{"1.2.0", false},
+		{"2.9.0", false},
+		{"2.0.0", false},
+		{"3.79.3", false},
+		{"3.80.0", true},
+		{utils.Development, true},
+		{"3.81.2", true},
+		{"4.15.2", true},
+	}
+	for _, test := range tests {
+		t.Run(test.xrayVersion, func(t *testing.T) {
+			result := isVersionCompatible(test.xrayVersion)
+			if test.expectedResult != result {
+				t.Error(fmt.Errorf("expected %t, got %t", test.expectedResult, result))
+			}
+		})
+	}
+}
+
+func TestXrayUsageEventToJson(t *testing.T) {
+	type reportUsageTestCase struct {
+		productId  string
+		EventId    string
+		Attributes []ReportUsageAttribute
+	}
+	jsonPatterns := []string{
+		`[{"product_name":"%s","event_name":"%s","origin":"API"}]`,
+		`[{"data":{"%s":"%s"},"product_name":"%s","event_name":"%s","origin":"API"}]`,
+		`[{"data":{"%s":"%s","%s":"%s"},"product_name":"%s","event_name":"%s","origin":"API"}]`,
+	}
+
+	cases := []reportUsageTestCase{
+		{"jfrog-cli-go", "generic_audit", []ReportUsageAttribute{}},
+		{"frogbot", "scan_pull_request", []ReportUsageAttribute{{AttributeName: "clientId", AttributeValue: "repo1"}}},
+		{"jfrog-idea-plugin", "ci", []ReportUsageAttribute{{AttributeName: "buildNumber", AttributeValue: "1023456"}, {AttributeName: "clientId", AttributeValue: "user-hash"}}},
+	}
```

Comment:
Let's merge the jsonPatterns into the test cases:
```go
type reportUsageTestCase struct {
	productId   string
	EventId     string
	Attributes  []ReportUsageAttribute
	jsonPattern string
}

cases := []reportUsageTestCase{
	{"jfrog-cli-go", "generic_audit", []ReportUsageAttribute{}, `[{"product_name":"%s","event_name":"%s","origin":"API"}]`},
	{"frogbot", "scan_pull_request", []ReportUsageAttribute{{AttributeName: "clientId", AttributeValue: "repo1"}}, `[{"data":{"%s":"%s"},"product_name":"%s","event_name":"%s","origin":"API"}]`},
	{"jfrog-idea-plugin", "ci", []ReportUsageAttribute{{AttributeName: "buildNumber", AttributeValue: "1023456"}, {AttributeName: "clientId", AttributeValue: "user-hash"}}, `[{"data":{"%s":"%s","%s":"%s"},"product_name":"%s","event_name":"%s","origin":"API"}]`},
}
```

Also for better method readability, I'd prefer to put the test cases outside of the test method.

Let's do this change also for TestEcosystemReportUsageToJson.